src/Raven.Server/Documents/Handlers/Processors/Attachments/RangeHelper.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Raven.Server.Documents.Handlers.Processors.Attachments
+{
+    internal static class RangeHelper
+    {
+        public static (bool SendBody, long Start, long? Length) SetRangeHeaders(HttpContext context, long size)
+        {
+            var responseHeaders = context.Response.GetTypedHeaders();
+            context.Response.Headers.AcceptRanges = "bytes";
+
+            long start = 0;
+            var (isRangeRequest, range) = ParseRange(context, size);
+            if (isRangeRequest)
+            {
+                if (range is null)
+                {
+                    context.Response.StatusCode = (int)HttpStatusCode.RequestedRangeNotSatisfiable;
+                    responseHeaders.ContentRange = new ContentRangeHeaderValue(size);
+                    responseHeaders.ContentLength = 0;
+                    return (false, start, null);
+                }
+
+                context.Response.StatusCode = (int)HttpStatusCode.PartialContent;
+                responseHeaders.ContentRange = ComputeContentRange(range, size, out start, out var length);
+                responseHeaders.ContentLength = length;
+                return (true, start, length);
+            }
+            else
+            {
+                responseHeaders.ContentLength = size;
+                return (true, 0, null);
+            }
+        }
+
+        private static (bool IsRangeRequest, RangeItemHeaderValue Range) ParseRange(HttpContext context, long length)
+        {
+            var rawRangeHeader = context.Request.Headers.Range;
+            if (StringValues.IsNullOrEmpty(rawRangeHeader))
+            {
+                return (false, null);
+            }
+
+            // Perf: Check for a single entry before parsing it
+            if (rawRangeHeader.Count > 1 || (rawRangeHeader[0] ?? string.Empty).Contains(','))
+            {
+                // The spec allows for multiple ranges but we choose not to support them because the client may request
+                // very strange ranges (e.g. each byte separately, overlapping ranges, etc.) that could negatively
+                // impact the server. Ignore the header and serve the response normally.
+                return (false, null);
+            }
+
+            var rangeHeader = context.Request.GetTypedHeaders().Range;
```

Comment:
You are using both `GetTypedHeaders` and `Headers` in this method, why?

Reply:
This code is taken from AspNetCore (unfortunately it's internal - https://github.com/dotnet/aspnetcore/blob/main/src/Shared/RangeHelper/RangeHelper.cs). 
The reason is a perf improvement because we can check if the range header is valid before we have to parse it with `GetTypedHeaders`. If you would prefer simplicity I can take it out and check the parsed value from TypedHeaders directly.

Reply:
No, that's fine given that you have a good reason for that